keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components/README.md

```diff
@@ -0,0 +1,239 @@
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP proposes to deprecate and in the future to remove a subset of the klog
+command line flags from Kubernetes components to encourage more diverse 
+approaches to logging in kubernetes ecosystem logging.
+
+## Motivation
+
+Early on Kubernetes adopted glog logging library for logging. Overtime the glog
+was forked to klog and multiple improvements were implemented, but features put
+into klog only piled up and were never removed. Introduction of alternative log
+formats like JSON created a conundrum, should we implement all klog features for
+JSON? Most of them don't make sense and method for their configuration leaves
+much to be desired. Klog features are controlled by set of global flags that
+remain last bastion of global state in k/k repository. Those flags don't have a
+single naming standard (some start with log prefix, some not), don't comply to
+k8s flag naming (use underscore instead of hyphen) and many other problems. We
+need to revisit how logging configuration is done in klog so it can work with
+alternative log formats and comply with current best practices.
+
+Large number of features added to klog has lead to large drop in quality. 
+[#90804](https://github.com/kubernetes/kubernetes/pull/90804) shows example 
+where kubeup (canonical way to deploy kubernetes for testing) could not use 
+klog feature to write log files due to scalability issues. The maintainers of 
+klog decided it's easier to re-implementing a canonical klog feature in external
+project than debugging the underlying problem.
+
+As for what configuration options should be standardized for all logging formats
+I would look into 12 factor app standard (https://12factor.net/). It defines
+logs as steams of events and discourages applications from taking on
+responsibility for log file management, log rotation and any other processing
+that can be done externally. This is something that Kubernetes already
+encourages by collecting stdout and stderr logs and making them available via
+kubectl logs. It's somewhat confusing that K8s components don't comply to K8s
+best practices.
+
+### Goals
+
+* Unblock development of alternative logging formats
+* Improve quality of logging in k8s core components
```

Comment:
I think we are missing one of the critical goals hinted at above, which is "Reduce complexity of klog configuration in Kubernetes components." I also think we're missing the "why" in "Unblock development of alternative logging formats" -- it's not clear why this is blocked right now, so I would add "by reducing the scope of required logging flags".

"Improve quality of logging" is too vague imo.